framework/decode/custom_vulkan_to_string.cpp

```diff
@@ -0,0 +1,443 @@
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+** FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+** DEALINGS IN THE SOFTWARE.
+*/
+
+#include "custom_vulkan_to_string.h"
+#include "generated/generated_vulkan_enum_to_string.h"
+#include "generated/generated_vulkan_struct_to_string.h"
+
+#include <algorithm>
+
+GFXRECON_BEGIN_NAMESPACE(gfxrecon)
+GFXRECON_BEGIN_NAMESPACE(util)
+
+// clang-format off
+
+// NOTE : The following structures' ToString() functions aren't generated due
+//  to various reasons...ie unions, non standard counts for arrays, fields that
+//  need validation to interpret correctly, etc...
+
+template <>
+std::string ToString<SECURITY_ATTRIBUTES>(const SECURITY_ATTRIBUTES& obj, ToStringFlags toStringFlags, uint32_t tabCount, uint32_t tabSize)
+{
+    return ObjectToString(toStringFlags, tabCount, tabSize,
+        [&](std::stringstream& strStrm)
+        {
+            FieldToString(strStrm, true, "nLength", toStringFlags, tabCount, tabSize, ToString(obj.nLength, toStringFlags, tabCount, tabSize));
+            FieldToString(strStrm, false, "lpSecurityDescriptor", toStringFlags, tabCount, tabSize, '"' + PtrToString(obj.lpSecurityDescriptor) + '"');
+            FieldToString(strStrm, false, "bInheritHandle", toStringFlags, tabCount, tabSize, ToString(obj.bInheritHandle, toStringFlags, tabCount, tabSize));
+        }
+    );
+}
+
+template <>
+std::string ToString<VkAccelerationStructureGeometryKHR>(const VkAccelerationStructureGeometryKHR& obj, ToStringFlags toStringFlags, uint32_t tabCount, uint32_t tabSize)
+{
+    return ObjectToString(toStringFlags, tabCount, tabSize,
+        [&](std::stringstream& strStrm)
+        {
+            FieldToString(strStrm, true, "sType", toStringFlags, tabCount, tabSize, ToString(obj.sType, toStringFlags, tabCount, tabSize));
+            FieldToString(strStrm, false, "pNext", toStringFlags, tabCount, tabSize, PNextToString(obj.pNext, toStringFlags, tabCount, tabSize));
+            FieldToString(strStrm, false, "geometryType", toStringFlags, tabCount, tabSize, ToString(obj.geometryType, toStringFlags, tabCount, tabSize));
+            FieldToString(strStrm, false, "geometry", toStringFlags, tabCount, tabSize, ToString(obj.geometry, toStringFlags, tabCount, tabSize));
+            FieldToString(strStrm, false, "flags", toStringFlags, tabCount, tabSize, ToString(obj.flags, toStringFlags, tabCount, tabSize));
+        }
+    );
+}
+
+template <>
+std::string ToString<VkAccelerationStructureGeometryDataKHR>(const VkAccelerationStructureGeometryDataKHR& obj, ToStringFlags toStringFlags, uint32_t tabCount, uint32_t tabSize)
+{
+    return ObjectToString(toStringFlags, tabCount, tabSize,
+        [&](std::stringstream& strStrm)
+        {
+            FieldToString(strStrm, true, "triangles", toStringFlags, tabCount, tabSize, ToString(obj.triangles, toStringFlags, tabCount, tabSize));
```

Comment:
VkAccelerationStructureGeometryDataKHR is union. Here, it prints every member. In Dx12, it only prints the used member.  I just wonder which format style is our prefer. @bradgrantham-lunarg

Reply:
I chose to print every member for all unions (except for `VkPerformanceValueDataINTEL` because one of them is a `char*`).  In my opinion it's simpler to print them all so that unions can be stringified without context, but I'm definitely willing to setup code that will drill into unions from structures that provide that context and only print the necessary field, if that's needed.